pkg/kubectl/cmd/apply_diff_last_applied.go

```diff
@@ -0,0 +1,62 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"io"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/kubernetes/pkg/kubectl"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/templates"
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/cmd/util/diff"
+	"k8s.io/kubernetes/pkg/util/i18n"
+)
+
+var (
+	applyDiffLastAppliedLong = templates.LongDesc(i18n.T(`
+		Opens up a 2-way diff in the default diff viewer. This should follow the same semantics as 'git diff'.
+		It should accept an environment variable KUBECTL_EXTERNAL_DIFF=meld to specify a custom diff tool.
```

Comment:
It should have flag `--diff-viewer` as described in the proposal.
It should take either `KUBECTL_EXTERNAL_DIFF` or `--diff-viewer`. If they are both specified, it should check if they match. If not, error out.